### ISSUE TITLE
Refactor

### ISSUE THREAD

**Issue by jancharatan** [sunil-2000/llm-puzzle-interfacing#4]
- We can make more stuff common between Connections + Wordle

**Comment by sunil-2000** [comment]
yes agree; abstract maybe at level of game: a game has an init, action, response, prepare (e.g., refreshing dom for next state), terminate (e.g., game over).
